languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -10069,6 +10069,34 @@ though, and has since been slightly modified)-->
     </rulegroup>
   </category>
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+      <rulegroup id="C_EST_QUOI" name="c'est quoi" default="temp_off">
+          <rule>
+              <pattern>
+                  <marker>
+                      <token>c'</token>
+                      <token>est</token>
+                      <token>quoi</token>
+                  </marker>
+                  <token postag="D e s|D f s" postag_regexp="yes"/>
+              </pattern>
+              <message>Cette expression peut être considérée comme familière.</message>
+              <suggestion>quelle est</suggestion>
+              <example correction="quelle est">Dis-moi <marker>c'est quoi</marker> la question.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <marker>
+                      <token>c'</token>
+                      <token>est</token>
+                      <token>quoi</token>
+                  </marker>
+                  <token postag="D e s|D m s" postag_regexp="yes"/>
+              </pattern>
+              <message>Cette expression peut être considérée comme familière.</message>
+              <suggestion>quel est</suggestion>
+              <example correction="quel est">Dis-moi <marker>c'est quoi</marker> ton nom.</example>
+          </rule>
```

Comment:
You always need to ask yourself "which obvious FPs is this pattern going to produce?"... even if the rule editor looks fine.

Problem with both rules:

- C'est quoi ce bordel ? → Quel est ce bordel ?
- C'est quoi ce bruit ? → Quel est ce bruit ?
- C'est quoi cette odeur ? → Quelle est cette odeur ?
- C'est quoi un adverbe ? → Quel est un adverbe ?
- C'est quoi une préposition ? → Quelle est une préposition ?
- C'est quoi le bonheur ? → Quel est le bonheur ?
- C'est quoi la philosophie ? → Quelle est la philosophie ?

And, by the way, making it case_sensitive or excluding SENT_START will not fix the problem. Users don't always start their sentences with an uppercase letter, and any of the above sentences might as well start with `Alors` or `Alors, `.

Reply:
Any feedback on this?

Reply:
Indeed, it was a tricky problem. I created subrules and I restricted the rule so it should be ready.